ui/src/AudioRecorder.js

```diff
@@ -0,0 +1,36 @@
+export class AudioRecorder {
+
+    constructor() {
+        this.recordingUrl = "";
+        this.mediaRecorder = "";
+        this.recordedChunks = [];
+    }
+
+    startRecording() {
+        navigator.mediaDevices.getUserMedia({ audio: true, video: false })
+            .then(function(stream) {
+
+            this.mediaRecorder = new MediaRecorder(stream);
+            this.mediaRecorder.start();
+
+        });
+    }
+
+    stopRecording() {
+        this.mediaRecorder.stop();
```

Comment:
Add a null check before calling methods on this.mediaRecorder because it might not always exist.